x-pack/plugins/lens/public/editor_frame_service/embeddable/expression_wrapper.tsx

```diff
@@ -37,6 +39,55 @@ export interface ExpressionWrapperProps {
   className?: string;
 }
 
+interface VisualizationErrorProps {
+  errors: ExpressionWrapperProps['errors'];
+}
+
+export function VisualizationErrorPanel({ errors }: VisualizationErrorProps) {
+  const defaultMessage = (
+    <EuiFlexItem>
+      <EuiText size="s">
+        <FormattedMessage
+          id="xpack.lens.embeddable.failure"
+          defaultMessage="Visualization couldn't be displayed"
+        />
+      </EuiText>
+    </EuiFlexItem>
+  );
+  // only showing one error as for now
+  const customError =
+    errors == null ? null : (
+      <EuiFlexItem className="eui-textBreakAll">{errors[0].longMessage}</EuiFlexItem>
+    );
+  // show that there are more errors, but it requires to open the editor to actually see them
+  const moreErrors =
+    errors && errors.length > 1 ? (
+      <EuiFlexItem>
+        <FormattedMessage
+          id="xpack.lens.embeddable.moreErrors"
+          defaultMessage="(Edit in Lens editor to see more errors)"
```

Comment:
Rather than having in parenthesis, can we treat this like a regular sentence?

Also, is it possible to make `Edit in Lens editor` a link that opens this visualization in Lens in a new tab? Offering the user a quick action to correct the problem would be nice.

Reply:
> Rather than having in parenthesis, can we treat this like a regular sentence?
> 
> Also, is it possible to make `Edit in Lens editor` a link that opens this visualization in Lens in a new tab? Offering the user a quick action to correct the problem would be nice.

Tried to implement the `Edit in Lens` button with the same logic of the Edit action (dropdown context menu) but it's not trivial. Perhaps it could be done as a follow up?

Reply:
Sure! Let me know if you'd like me to open an issue for it.